src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -69,9 +69,9 @@ public override bool CanConvert(Type typeToConvert)
 
         internal override ConverterStrategy ConverterStrategy => ConverterStrategy.Value;
 
-        internal sealed override JsonPropertyInfo CreateJsonPropertyInfo(JsonTypeInfo parentTypeInfo)
+        internal sealed override JsonPropertyInfo CreateJsonPropertyInfo(JsonTypeInfo declaringTypeInfo, Type propertyType, JsonSerializerOptions options)
         {
-            return new JsonPropertyInfo<T>(parentTypeInfo);
+            return new JsonPropertyInfo<T>(declaringTypeInfo.Type, propertyType, declaringTypeInfo, options);
```

Comment:
should we just take `declaringTypeInfo` rather than both `declaringType` and `declaringTypeInfo`? Is there ever a situation when `declaringType != declaringTypeInfo.Type`? I think when it's null they're both null (PropertyInfoForTypeInfo)

Reply:
The additional parameter exists for cases where `declaringTypeInfo` is null.